### PR TITLE
MVSortingView and GUI styling

### PR DESCRIPTION
**Pull request by magland**
A bunch of CSS tweaks, color adjustments and explicit layout adjustments. Also, MVSortingView is now displayed for SortingView as opposed to the accordion view. Note that all the plugin views are available within MVSortingView, so we don't need both.

Screenshot:
image

### CODE REVIEW

File: jupyterlab/labbox_ephys_widgets_jp/src/pluginWidgets.tsx [hunk 1]
```diff
@@ -162,8 +162,8 @@ export class SortingViewModel extends DOMWidgetModel {
       pluginName: '',
       sortingObject: {},
       recordingObject: {},
-      curation: defaultSortingCuration,
-      selection: defaultSortingSelection
+      curation: {},
```
Comment (magland) [inline jupyterlab/labbox_ephys_widgets_jp/src/pluginWidgets.tsx:165]: switch to new defaults for selection/curation in jupyter extension

File: src/AppContainer.js [hunk 2]
```diff
@@ -50,9 +52,36 @@ const SetDocumentInfo = ({ documentId, feedUri, onSetDocumentInfo }) => {
     return <div>Setting document info...</div>
 }
 
-const AppContainer = ({ location, initialLoadComplete, children, documentInfo, onSetDocumentInfo, extensionsConfig }) => {
+// Thanks: https://stackoverflow.com/questions/36862334/get-viewport-window-height-in-reactjs
```
Comment (magland) [inline src/AppContainer.js:55]: The app bar is now sticky at the top and scrollbar controls the main content. We need to pass the browser window width/height to the MVSorting view, so we use this custom hook to get the browser window dimensions. This width/height state is changed whenever the browser is resized (very convenient).

File: src/AppContainer.js [hunk 3]
```diff
@@ -70,19 +99,25 @@ const AppContainer = ({ location, initialLoadComplete, children, documentInfo, o
...
                 <Toolbar>
                     <ToolBarContent documentInfo={documentInfo} extensionsConfig={extensionsConfig} />
                 </Toolbar>
             </AppBar>
-            <div style={{padding: 30}}>
+            <div classNAme={"AppContent"} style={{padding: 0, position: 'absolute', top: toolBarHeight, height: H, left: hMargin, width: W, overflowY: 'auto', overflowX: 'hidden'}}>
                 {
                     initialLoadComplete ? (
-                        children
+                        <Routes width={W} height={H} />
```
Comment (magland) [inline src/AppContainer.js:120]: We are now more explicit about passing width/height to child components.

File: src/Routes.tsx [hunk 4]
```diff
@@ -55,7 +55,7 @@ const Routes: FunctionComponent<{}> = () => {
             <Route
                 path="/:documentId/sorting/:sortingId*"
                 render={({ match }) => (
-                    <SortingView sortingId={match.params.sortingId} />
+                    <SortingView sortingId={match.params.sortingId} width={width} height={height} />
```
Comment (magland) [inline src/Routes.tsx:58]: Explicitly pass the computed width/height

File: src/containers/SortingView.tsx [hunk 5]
```diff
@@ -181,7 +181,10 @@ const SortingView: React.FunctionComponent<Props> = (props) => {
     return <h3>{`Recording not found: ${sorting.recordingId}`}</h3>
   }
 
-  const selectedUnitIdsLookup: {[key: string]: boolean} = selection.selectedUnitIds.reduce((m, uid) => {m[uid + ''] = true; return m}, {} as {[key: string]: boolean})
+  const sv = plugins.sortingViews['MVSortingView']
```
Comment (magland) [inline src/containers/SortingView.tsx:184]: Now the SortingView displays only the MVSortingView as opposed to showing the accordion view of the plugins. Note that those plugin views are all available within the MVSortingView.

File: src/containers/SortingView.tsx [hunk 6]
```diff
@@ -181,7 +181,10 @@ const SortingView: React.FunctionComponent<Props> = (props) => {
...
+  if (!sv) throw Error('Missing sorting view: MVSortingView')
```
Comment (magland) [inline src/containers/SortingView.tsx:185]: We access the MVSortingView through the plugin/extension system, and throw an error if not found.

File: src/containers/SortingView.tsx [hunk 7]
```diff
@@ -190,7 +193,7 @@ const SortingView: React.FunctionComponent<Props> = (props) => {
             {recording.recordingLabel}
           </Link>
       </h3>
-      {
+      {/* {
```
Comment (magland) [inline src/containers/SortingView.tsx:196]: Removing the left panel that used to show the units.

File: src/containers/SortingView.tsx [hunk 8]
```diff
@@ -269,12 +287,21 @@ const mapStateToProps: MapStateToProps<StateProps, OwnProps, RootState> = (state
...
+    if (a.type === 'SetCuration') {
+      dispatch({type: 'SET_CURATION', sortingId: ownProps.sortingId, curation: a.curation, persistKey: 'sortings'})
+    }
+    else if (a.type === 'AddLabel') {
       dispatch({type: 'ADD_UNIT_LABEL', sortingId: ownProps.sortingId, unitId: a.unitId, label: a.label, persistKey: 'sortings'})
     }
     else if (a.type === 'RemoveLabel') {
       dispatch({type: 'REMOVE_UNIT_LABEL', sortingId: ownProps.sortingId, unitId: a.unitId, label: a.label, persistKey: 'sortings'})
     }
+    else if (a.type === 'MergeUnits') {
```
Comment (magland) [inline src/containers/SortingView.tsx:299]: I don't know why these changes are showing up in this PR... this should already be merged into master. Hmmm.

File: src/extensions/common/NiceTable.css [hunk 9]
```diff
@@ -1,11 +1,11 @@
-.NiceTable .MuiTableCell-root {
+/* .NiceTable .MuiTableCell-root {
```
Comment (magland) [inline src/extensions/common/NiceTable.css:1]: Reorganizing how css gets handled

File: src/extensions/mountainview/MVSortingView/MVSortingView.tsx [hunk 10]
```diff
@@ -1,8 +1,9 @@
 import React, { FunctionComponent, useCallback, useEffect, useReducer } from 'react'
 import sizeMe, { SizeMeProps } from 'react-sizeme'
 import Expandable from "../../curation/CurationSortingView/Expandable"
-import { defaultSortingCuration, SortingUnitViewPlugin, SortingViewPlugin, SortingViewProps, ViewPlugin } from "../../extensionInterface"
+import { SortingUnitViewPlugin, SortingViewPlugin, SortingViewProps, ViewPlugin } from "../../extensionInterface"
 import Splitter from '../../timeseries/TimeWidgetNew/Splitter'
+import '../mountainview.css'
```
Comment (magland) [inline src/extensions/mountainview/MVSortingView/MVSortingView.tsx:6]: We import mountainview.css which is the styling specific to MVSortingView, etc.

File: src/extensions/mountainview/MVSortingView/MVSortingView.tsx [hunk 11]
```diff
@@ -117,49 +118,51 @@ const MVSortingView: FunctionComponent<SortingViewProps & SizeMeProps> = (props)
             area
         })
     }, [openViewsDispatch])
-    const width = size.width || 600
-    const height = 900 // hard-coded for now
+    const width = props.width || size.width || 600
```
Comment (magland) [inline src/extensions/mountainview/MVSortingView/MVSortingView.tsx:121]: Get the width/height explicitly from the parent.

File: src/extensions/mountainview/MVSortingView/ViewContainer.tsx [hunk 12]
```diff
@@ -11,7 +11,8 @@ type Props = {
     height: number
 }
 
-const tabBarHeight = 60
+// needs to correspond to css (not best system)
+const tabBarHeight = 40 + 5
```
Comment (magland) [inline src/extensions/mountainview/MVSortingView/ViewContainer.tsx:15]: Misc styling / position / size adjustments throughout this file.

File: src/extensions/mountainview/MVSortingView/ViewContainerTabBar.tsx [hunk 13]
```diff
@@ -8,9 +8,10 @@ type Props = {
     currentView: View | null
     onCurrentViewChanged: (v: View) => void
     onViewClosed: (v: View) => void
+    active: boolean
 }
 
-const ViewContainerTabBar: FunctionComponent<Props> = ({ views, currentView, onCurrentViewChanged, onViewClosed }) => {
+const ViewContainerTabBar: FunctionComponent<Props> = ({ views, currentView, onCurrentViewChanged, onViewClosed, active }) => {
```
Comment (magland) [inline src/extensions/mountainview/MVSortingView/ViewContainerTabBar.tsx:14]: Style the tabs based on whether they are selected and whether they are in a group (north/south) that is active

File: src/extensions/mountainview/mountainview.css [hunk 14]
```diff
@@ -0,0 +1,40 @@
+.MVSortingView .ViewContainerTabBar {
```
Comment (magland) [inline src/extensions/mountainview/mountainview.css:1]: Here's the new styling for MVSortingView

File: src/extensions/unitstable/Units/UnitsTable.tsx [hunk 15]
```diff
@@ -2,6 +2,7 @@ import { Checkbox, LinearProgress, Table, TableBody, TableCell, TableHead, Table
 import React, { FunctionComponent, useCallback, useState } from 'react';
 import { ExternalSortingUnitMetric, Sorting, SortingSelection, SortingSelectionDispatch, SortingUnitMetricPlugin } from '../../extensionInterface';
 import sortByPriority from '../../sortByPriority';
+import '../unitstable.css';
```
Comment (magland) [inline src/extensions/unitstable/Units/UnitsTable.tsx:5]: unitstable.css is the styling specific to the units table

File: src/index.tsx [hunk 16]
```diff
@@ -203,9 +202,7 @@ const content = (
   <MuiThemeProvider theme={theme}>
     <Provider store={theStore}>
       <Router>
-        <AppContainer>
-          <Routes />
-        </AppContainer>
+        <AppContainer />
```
Comment (magland) [inline src/index.tsx:205]: The <Routes /> is now in the AppContainer file, where the width and height are set explicitly.

**Review by jsoules — APPROVED** [review]
No real comments here--I think we can go ahead and merge.
